mindmeld/components/dialogue.py

```diff
@@ -866,13 +866,21 @@ def _validate(self, request, slot):
                 return False, _resolved_value
 
         if slot.custom_eval:
-            # Custom validation using function provided by developer. Should return True/False
-            # for validation status. If true, then continue, else fail overall validation.
-
-            if not slot.custom_eval(request):
+            # Custom validation using function provided by developer.
+            # Should return True/False/Custom Resolution value for validation status.
+            # If false, overall validation fails. If either true or a custom resolved
+            # value is returned, then the validation succeeds.
+
+            _validity = slot.custom_eval(request)
+            if not _validity:
+                # For checking 'false' return cases
```

Comment:
We should check `is False`, so that the developer can use falsy values like `0` and `None` as valid slot values.

Reply:
Done!